misk/src/test/kotlin/misk/web/ContentBasedDispatchTest.kt

```diff
@@ -130,4 +195,19 @@ internal class ContentBasedDispatchTest {
         assertThat(response.code()).isEqualTo(200)
         return response.body()!!
     }
-}
\ No newline at end of file
+
+    class ComplexPacketUnmarshaller : Unmarshaller {
+        override fun unmarshal(source: BufferedSource): Any?
+            = ComplexPacket(source.readUtf8(), "unmarshaller")
+
+        class Factory: Unmarshaller.Factory {
+            override fun create(mediaType: MediaType, type: KType): Unmarshaller? {
+                if (mediaType.type() != complexMediaType.type() ||
+                    mediaType.subtype() != complexMediaType.subtype()) return null
+
+                if (GenericUnmarshallers.canHandle(type)) return null
```

Comment:
this one makes me anxious. . . I think this precedence call should be somewhere else . . .if this marshaller is lower-precedence than the generic ones then it should be called later

Reply:
This call doesn't really make sense now that I think of it; if GenericUnmarshallers can do a better job of converting that type why not just do it directly? I got this pattern from Json unmarshalling, seemed like a good idea when not thinking too much about it.